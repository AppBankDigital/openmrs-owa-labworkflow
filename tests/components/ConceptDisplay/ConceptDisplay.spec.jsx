import React from 'react';
import { Provider } from 'react-redux';
import { mountWithIntl } from '@openmrs/react-components';
import ConceptDisplay from '../../../app/js/components/ConceptDisplay';


let mountedComponent;


const getComponent = (state, props) => {
  const store = mockStore(state);

  if (!mountedComponent) {
    mountedComponent = mountWithIntl(
      <Provider store={store}>
        <ConceptDisplay {...props} />
      </Provider>
      ,
    );
  }

  return mountedComponent;
};

describe('ConceptDisplay Component', () => {
  it('should render for concept value of type result', (done) => {
    const props = {
      conceptUUID: 'some-concept-uuid',
      type: 'result',
      value: 30,
    };

    const state = {
      openmrs: {
        session: {
          locale: 'en'
        }
      },
      conceptMembers: {
        'some-concept-uuid': {
          hiNormal: 50,
          lowNormal: 40,
          units: 'mmgh',
        },
      },
    };
    const component = getComponent(state, props);
    expect(component).toMatchSnapshot();
    done();
  });

  it('should render for concept value of type range when there is valid units', (done) => {
    const props = {
      conceptUUID: 'some-concept-uuid',
      type: 'range',
      value: 30,
    };

    const state = {
      conceptMembers: {
        'some-concept-uuid': {
          hiNormal: 50,
          lowNormal: 40,
          units: 'mmgh',
        },
      },
    };

    const component = getComponent(state, props);
    expect(component).toMatchSnapshot();
    done();
  });

  it('should render for concept value of type range when unit is null', (done) => {
    const props = {
      conceptUUID: 'some-concept-uuid',
      type: 'range',
      value: 30,
    };

    const state = {
      conceptMembers: {
        'some-concept-uuid': {
          hiNormal: 50,
          lowNormal: 40,
        },
      },
    };

    const component = getComponent(state, props);
    expect(component).toMatchSnapshot();
    done();
  });
});
