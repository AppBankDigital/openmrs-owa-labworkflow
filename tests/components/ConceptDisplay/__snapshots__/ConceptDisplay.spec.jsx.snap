// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConceptDisplay Component should render for concept value of type range when there is valid units 1`] = `
<Provider
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "textComponent": "span",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ConceptDisplay)
    conceptUUID="some-concept-uuid"
    type="result"
    value={30}
  >
    <ConceptDisplay
      concept={
        Object {
          "hiNormal": 50,
          "lowNormal": 40,
          "units": "mmgh",
        }
      }
      conceptMembers={
        Object {
          "some-concept-uuid": Object {
            "hiNormal": 50,
            "lowNormal": 40,
            "units": "mmgh",
          },
        }
      }
      conceptUUID="some-concept-uuid"
      dispatch={[Function]}
      type="result"
      value={30}
    >
      <span
        className="abnormal-value"
      >
        30
      </span>
    </ConceptDisplay>
  </Connect(ConceptDisplay)>
</Provider>
`;

exports[`ConceptDisplay Component should render for concept value of type range when unit is null 1`] = `
<Provider
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "textComponent": "span",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ConceptDisplay)
    conceptUUID="some-concept-uuid"
    type="result"
    value={30}
  >
    <ConceptDisplay
      concept={
        Object {
          "hiNormal": 50,
          "lowNormal": 40,
          "units": "mmgh",
        }
      }
      conceptMembers={
        Object {
          "some-concept-uuid": Object {
            "hiNormal": 50,
            "lowNormal": 40,
            "units": "mmgh",
          },
        }
      }
      conceptUUID="some-concept-uuid"
      dispatch={[Function]}
      type="result"
      value={30}
    >
      <span
        className="abnormal-value"
      >
        30
      </span>
    </ConceptDisplay>
  </Connect(ConceptDisplay)>
</Provider>
`;

exports[`ConceptDisplay Component should render for concept value of type result 1`] = `
<Provider
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "textComponent": "span",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ConceptDisplay)
    conceptUUID="some-concept-uuid"
    type="result"
    value={30}
  >
    <ConceptDisplay
      concept={
        Object {
          "hiNormal": 50,
          "lowNormal": 40,
          "units": "mmgh",
        }
      }
      conceptMembers={
        Object {
          "some-concept-uuid": Object {
            "hiNormal": 50,
            "lowNormal": 40,
            "units": "mmgh",
          },
        }
      }
      conceptUUID="some-concept-uuid"
      dispatch={[Function]}
      type="result"
      value={30}
    >
      <span
        className="abnormal-value"
      >
        30
      </span>
    </ConceptDisplay>
  </Connect(ConceptDisplay)>
</Provider>
`;
